### PR TITLE
Fix a number of warnings in ems.cpp

### PR DESCRIPTION
**Pull request by dreamer**
effc++ fixes are trivial, but the second commit warrants attention. I left detailed explanation in the commit message.

### CODE REVIEW

File: src/ints/ems.cpp [hunk 1]
```diff
@@ -630,15 +631,23 @@ static Bit8u MemoryRegion(void) {
...
+
 	Bitu toread;
 	while (region.bytes>0) {
-		if (region.bytes>MEM_PAGE_SIZE) toread=MEM_PAGE_SIZE;
-		else toread=region.bytes;
+		if (region.bytes > MEM_PAGE_SIZE)
+			toread = MEM_PAGE_SIZE;
+		else
+			toread = region.bytes;
+
```
Comment (kcgen) [inline src/ints/ems.cpp:644]: Might be able to swap in: `toread = std::min(region.bytes, MEM_PAGE_SIZE);`

File: src/ints/ems.cpp [hunk 2]
```diff
@@ -630,15 +632,19 @@ static Bit8u MemoryRegion(void) {
...
 		/* Read from the source */
 		if (!region.src_type) {
 			MEM_BlockRead(src_mem,buf_src,toread);
 		} else {
-			if (toread<src_remain) {
+			assert(toread <= MEM_PAGE_SIZE);
+			if (toread <= src_remain) {
 				MEM_BlockRead((src_handle*MEM_PAGE_SIZE)+src_off,buf_src,toread);
 			} else {
 				MEM_BlockRead((src_handle*MEM_PAGE_SIZE)+src_off,buf_src,src_remain);
```
Comment (kcgen) [inline src/ints/ems.cpp:650]: I was hoping we could keep collapsing all of these if / else blocks:
```
if (toread <= src_remain) {
	MEM_BlockRead((src_handle*MEM_PAGE_SIZE)+src_off,buf_src,toread);
} else {
	MEM_BlockRead((src_handle*MEM_PAGE_SIZE)+src_off,buf_src,src_remain);
```

into:
`MEM_BlockRead((src_handle*MEM_PAGE_SIZE)+src_off,buf_src, std::min(toread, src_remain));

But then expanded the diff to see the remainder handed with a subsequent call.  Oh well :-)

Comment (dreamer) [reply src/ints/ems.cpp:650]: Yeah, I spent some time in the morning collapsing this into a separate function… The result was a function with 8 parameters and 1 template parameter… Turned out to not be more readable at all - it only shifted complexity from one place in code to another.

Half of the variables in this function should probably be collapsed into a second temporary MoveRegion variable or something… not sure. With the second attempt, I decided to limit scope only to lines directly relevant to the Coverity finding.

**Comment by dreamer** [comment]
Thanks :) This should eliminate the last 2 "high impact" Coverity findings in `ints` module.

**Pull request merged**